shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -994,3 +957,56 @@ - (void)postMainThreadTask:(FlutterTask)task targetTimeInNanoseconds:(uint64_t)t
 }
 
 @end
+
+@implementation FlutterEngineRegistrar {
+  NSString* _pluginKey;
+  FlutterEngine* _flutterEngine;
+}
+
+- (instancetype)initWithPlugin:(NSString*)pluginKey flutterEngine:(FlutterEngine*)flutterEngine {
+  self = [super init];
+  if (self) {
+    _pluginKey = [pluginKey copy];
```

Comment:
Huh, the old code was missing a `dealloc` `-release`? Can you add it?

Reply:
The macOS embedding is all ARC.

Reply:
Oh oh right sorry.